django-verdant/rca/templates/rca/programme_page.html

```diff
@@ -62,18 +62,26 @@ <h2 class="module-title module-title-black offset-title">About the programme</h2
                                     {% endif %}
                                 </div>
                             </div>
-
-                            <h4 class="programme-head">Head{% if self.head_of_programme_second %}s{% endif %} of Programme</h4>
+                            {% if not self.head_of_programme_title and not self.head_of_programme_second_title and self.head_of_programme and self.head_of_programme_second %}
+                                <h4 class="programme-head">Heads of Programme</h4>
+                            {% else %}
+                                <h4 class="programme-head">{{ self.head_of_programme_title|default:'Head of Programme' }}</h4>
+                            {% endif %}
                             <ul>
                                 <li class="programme-head-name"><a href="{% pageurl self.head_of_programme %}" class="a3">{{ self.head_of_programme.title }}</a></li>
-                                {% if self.head_of_programme_second %}
-                                    <li class="programme-head-name"><a href="{% pageurl self.head_of_programme_second %}" class="a3">{{ self.head_of_programme_second.title }}</a></li>
-                                {% endif %}
                             </ul>
+                            {% if self.head_of_programme_second %}
+                                {% if self.head_of_programme_second_title or self.head_of_programme_title %}
+                                    <h4 class="programme-head">{{ self.head_of_programme_second_title|default:'Head of Programme' }}</h4>
+                                {% endif %}
+                                <ul>
+                                    <li class="programme-head-name"><a href="{% pageurl self.head_of_programme_second %}" class="a3">{{ self.head_of_programme_second.title }}</a></li>
+                                </ul>
+                            {% endif %}
```

Comment:
As a general point for readability, once logic gets this convoluted I'd consider moving it to a pair of methods `.get_head_of_programme_title()` and `.get_head_of_programme_second_title()`, and working out the cases in Python. E.g.

``` python
def get_head_of_programme_title(self): 
    return (
        self.head_of_programme_title
        or (
                # if there's also a second head with no title
                self.head_of_programme_second
                and not self.head_of_programme_second_title
                and "Heads of Programme"
        ) or "Head of Programme"

def get_head_of_programme_second_title(self):
    """ NB returns False if not
            self.head_of_programme_title or self.head_of_programme_second_title
    """"
    return (
        self.head_of_programme_second_title
        or ( self.head_of_programme_title and "Head of Programme" )
    )
```

This way the Python can be commented, and the template looks neater. Admittedly there are multiple ways of doing this, and you may prefer something else.

Reply:
Thanks @nimasmi! I took the point about implementing it in Python for greater readability, but I used slightly more simplistic looking code (subjectively).